src/platform/forms-system/src/js/definitions/address.js

```diff
@@ -258,9 +257,9 @@ export function uiSchema(
         if (isRequired) {
           const required = isRequired(formData, index);
           if (required && currentSchema.required.length === 0) {
-            currentSchema = _.set('required', requiredFields, currentSchema);
+            currentSchema = set('required', requiredFields, currentSchema);
```

Comment:
```suggestion
            currentSchema.required = requiredFields;
```

Reply:
Becasue `currentSchema` equals `addressSchema` from line 256, updating `currentSchema` like this would mutate `addressSchema`. Does that matter here?

Reply:
`set` is mutating it anyway. From [the docs](https://lodash.com/docs/4.17.15#set):

> **Note:** This method mutates object.

That's why I'm not a huge fan of it - it mutates _and_ returns the mutated object. It makes things harder to change because you don't know which one (or both) are the necessary parts.

Reply:
It's the `lodash` version that mutates the object. The `lodash/fp` [version](https://gist.github.com/jfmengels/6b973b69c491375117dc#_setpath-value-object) doesn't mutate the object. That's the version being used in this file. By swapping `lodash/fp`'s `set` without our platform `set` function, we are keeping the same functionality of not mutating the object. If that doesn't matter, committing the suggestion shouldn't be a problem though.